python/pyspark/sql/streaming/python_streaming_source_runner.py

```diff
@@ -76,6 +97,19 @@ def commit_func(reader: DataSourceStreamReader, infile: IO, outfile: IO) -> None
     write_int(0, outfile)
 
 
+def send_batch_func(
+    rows: Iterator[Tuple], outfile: IO, schema: StructType, data_source: DataSource
+) -> None:
+    batches = list(records_to_arrow_batches(rows, 1000, schema, data_source))
```

Comment:
Should the max_records_arrow_batch be set from a controllable Spark Config, rather than hardcoded value of 1000?

Reply:
Makes sense. Propagate the config from JVM now.